### PR TITLE
[EDT-1538] Next.ConnectorGrafxRegistration added to avoid breaking changes

### PR DESCRIPTION
**Pull request by evan-mcgeek**
This PR helps breaking changes with the engine, which stated to use `RemoteConnectorId` instead of `url` for `grafx` source connectors. Proposed types are suggested in the next initiator

## PR Guidelines

- [x] I have read the [contribution and PR guidelines](/chili-publish/studio-sdk/blob/main/CONTRIBUTING.md)

## Related tickets

-   [EDT-1538](https://chilipublishintranet.atlassian.net/browse/EDT-1538)

[EDT-1538]: https://chilipublishintranet.atlassian.net/browse/EDT-1538?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/sdk/src/types/ConnectorTypes.ts
Comment (brapoprod) [inline packages/sdk/src/types/ConnectorTypes.ts:78]: how did this even not break any tests 🧐

Comment (evan-mcgeek) [reply packages/sdk/src/types/ConnectorTypes.ts:78]: I guess, because `ConnectorUrlRegistration` is used in `ConnectorController` and `VariableController` tests.

Comment (evan-mcgeek) [reply packages/sdk/src/types/ConnectorTypes.ts:78]: https://github.com/chili-publish/studio-sdk/blob/main/packages/sdk/src/tests/controllers/ConnectorController.test.ts#L50-L53

https://github.com/chili-publish/studio-sdk/blob/main/packages/sdk/src/tests/controllers/VariableController.test.ts#L250-L253

File: packages/sdk/src/types/ConnectorTypes.ts
Comment (NicolaVerbeeck) [inline packages/sdk/src/types/ConnectorTypes.ts:59]: Accessible instead of available

File: packages/sdk/src/types/ConnectorTypes.ts
Comment (NicolaVerbeeck) [inline packages/sdk/src/types/ConnectorTypes.ts:70]: Add url field and mark it as deprecated so we are not strictly breaking

Comment (Matthiee) [reply packages/sdk/src/types/ConnectorTypes.ts:70]: Won't this change it from a compile-time error to a runtime error?

Comment (evan-mcgeek) [reply packages/sdk/src/types/ConnectorTypes.ts]: outdated :)

**Review by Matthiee — CHANGES_REQUESTED** [review]

**Review by Matthiee — DISMISSED** [review]

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:71]: Stable version of controller shouldn't import anything from `next` - instead you have to create a `next` version of controller and use a new type of params (Similar to variable controller)

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:38]: To make property private
```suggestion
    #connectorCompatibilityTools: ConnectorCompatibilityTools;

```

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:64]:
```suggestion
                        update.parsedData,

```

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:59]:
```suggestion
            .then((result) => getEditorResponseData<Next.ConnectorInstance>(result))

```

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:82]:
```suggestion
            .then((result) => getEditorResponseData<Next.ConnectorInstance>(result))

```

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (psamusev) [inline packages/sdk/src/controllers/ConnectorController.ts:87]:
```suggestion
                        update.parsedData,

```

File: packages/sdk/src/controllers/ConnectorController.ts
Comment (Matthiee) [inline packages/sdk/src/controllers/ConnectorController.ts:65]: I did a quick code search for this `chiliEnvironmentUrl` and found nothing using it. 

If this is merged, will it _**actually**_ work? 

Maybe @psamusev or @brapoprod knows.

Comment (pietervp) [reply packages/sdk/src/controllers/ConnectorController.ts:65]: Moreover, `makeSingleConnectorBackwardsCompatible` scans this `chiliEnvironmentUrl` for a connector Id, which will most certainly not work in this case?

Comment (psamusev) [reply packages/sdk/src/controllers/ConnectorController.ts:65]: > If this is merged, will it actually work?
We will check it on QA stage, before merge, to ensure

File: packages/sdk/src/utils/LocalConfigurationDecorator.ts
Comment (Matthiee) [inline packages/sdk/src/utils/LocalConfigurationDecorator.ts:5]: Should probably extend `ConfigurationController` or implement a common interface

File: packages/sdk/src/utils/LocalConfigurationDecorator.ts
Comment (Matthiee) [inline packages/sdk/src/utils/LocalConfigurationDecorator.ts:31]: An undefined check would be better

File: packages/sdk/src/utils/LocalConfigurationDecorator.ts
Comment (Matthiee) [inline packages/sdk/src/utils/LocalConfigurationDecorator.ts:53]: Only store it locally if the engine responds with success

File: packages/sdk/src/utils/LocalConfigurationDecorator.ts
Comment (evan-mcgeek) [reply packages/sdk/src/utils/LocalConfigurationDecorator.ts:31]: it is undefined check or WDYM?

Comment (Matthiee) [reply packages/sdk/src/utils/LocalConfigurationDecorator.ts:31]: `value` can contain an empty string which is also valid

Comment (psamusev) [reply packages/sdk/src/utils/LocalConfigurationDecorator.ts:31]:
```suggestion
        if (value != null) {
```

File: packages/sdk/src/sdk.ts
Comment (Matthiee) [inline packages/sdk/src/sdk.ts:65]: Consider hiding this implementation detail.

```suggestion
    configuration: ConfigurationController;
```

**Review by Matthiee — DISMISSED** [review]

**Review by Matthiee — APPROVED** [review]

**Review by brapoprod — CHANGES_REQUESTED** [review]
I'll block this PR until we're sure it's not breaking integrations @sebastianpujina will follow up

**Review by psamusev — APPROVED** [review]

**Review by brapoprod — APPROVED** [review]

**Pull request merged**
